### PR TITLE
Sites Management Page: Update status and sorting dropdown labels

### PR DESCRIPTION
**Pull request by zaguiini**
#### Proposed Changes

In order to improve accessibility, let's add labels so it's clear for screen readers that you're touching a dropdown with a value instead of just a button.

image

#### Testing Instructions

Enable the screen reader and tab through the dropdowns. The spoken message should match the intent of the button, which should indicate that the user is touching a select box.

### CODE REVIEW

**Review by vykes-mac — APPROVED** [review]
Good to go!

**Comment by wojtekn** [comment]
@zaguiini, this works as expected - I can read the field label now, and the screen reader reads it.

However, I noticed a few other issues related to the screen reader:
- when I open the "Status" dropdown using enter key, I can use arrows to navigate through options. I can't do the same in the "Sorting by" dropdown. Tab works in both, though.
- screen reader reads the number of sites without the context. Should it read "thirteen sites" instead of just "thirteen"?
- each option in the "Status" dropdown is read twice giving the confusing "Public Public thirteen", "Coming soon Coming soon two" etc.

Do we want to include those and other improvements for the screen readers in the scope of this ticket, or should we open another ticket?

File: client/sites-dashboard/components/sites-sorting-dropdown.tsx [hunk 1]
```diff
@@ -65,13 +65,13 @@ export const SitesSortingDropdown = ( {
 
 		switch ( sitesSorting ) {
 			case `lastInteractedWith${ SEPARATOR }desc`:
-				return __( 'Sorted automagically' );
+				return __( 'Sorting by: Automagically' );
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-sorting-dropdown.tsx]:
```suggestion
				return __( 'Sorting by: Magic' );
```

I _think_ this is more correct.

Comment (wojtekn) [reply client/sites-dashboard/components/sites-sorting-dropdown.tsx]: Or **Automaggic** 🤣

**Comment by SaxonF** [comment]
Looks great, thanks @zaguiini !

**Comment by danielbachhuber** [comment]
> However, I noticed a few other issues related to the screen reader:
>
> [...] 
>
> Do we want to include those and other improvements for the screen readers in the scope of this ticket, or should we open another ticket?

@wojtekn Nice finds! In the interest of moving this pull request along, I've captured those to a separate issue: https://github.com/Automattic/wp-calypso/issues/68021

**Review by danielbachhuber — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7539787

Thank you @zaguiini for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7539787).

**Pull request merged**